pyproject.toml

```diff
@@ -1,261 +1,266 @@
+pyaml-env = "*"            # Reading YAML output config files, with paths
+pyorbital = "*"            # required by satpy
+pypublicdecompwt = "*"     # Required to read SEVIRI data
+rasterio = "*"             # GEOTIFF output
+referencing = "*"
+satpy = "*"               # efficiency improvements >= 0.33.1
+scikit-image = "*"        # Radius based center coverage checks
+
+# Optional dependencies
+# To add a new optional dependency it must be added both here and in one of the "extra"
+# groups below.
+#
+# Doc group
+pydata-sphinx-theme = { version = "*", optional = true }
+sphinx = { version = "<7.2", optional = true }
+sphinx-design = { version = "*", optional = true }
+# Lint group
+bandit = { version = "*", optional = true }
+black = { version = "*", optional = true }
+pylint = { version = "*", optional = true }
+flake8 = { version = "*", optional = true }
+flake8-rst-docstrings = { version = "*", optional = true }
+flake8-docstrings = { version = "*", optional = true }
+flake8-rst = { version = "*", optional = true }
+prettier = { version = "*", optional = true }
+doc8 = { version = "*", optional = true }
+# Test group
+pytest = { version = "*", optional = true }
+pytest-cov = { version = "*", optional = true }
+pixelmatch = { version = "*", optional = true }
+# Debug group
+ipython = { version = "*", optional = true }
+
+#[tool.poetry.dev-dependencies]
+# nothing at this stage;
+
+[tool.poetry.extras]
 doc = [
-    "sphinx<7.2",  # Required for building documentation, v7.2 errors
-    "sphinx-design",  # Required for building documentation
-    "pydata-sphinx-theme",  # Required for building documentation
+    "pydata-sphinx-theme", # Required for building documentation
+    "sphinx",              # Required for building documentation, v7.2 errors
+    "sphinx-design"        # Required for building documentation
 ]
 lint = [
-    "bandit",  # Syntax/security checking
-    "black",   # Syntax checking and reformatting
-    "pylint",  # Syntax checking
-    "flake8",  # Syntax checking
-    "flake8-rst-docstrings",  # numpy docstring checking
-    "flake8-docstrings",  # docstring checking
-    "flake8-rst",  # RST checking
-    "prettier",  # YAML, JSON, etc linting
-    "doc8",  # RST linting
+    "bandit",                # Syntax/security checking
+    "black",                 # Syntax checking and reformatting
+    "pylint",                # Syntax checking
+    "flake8",                # Syntax checking
+    "flake8-rst-docstrings", # numpy docstring checking
+    "flake8-docstrings",     # docstring checking
+    "flake8-rst",            # RST checking
+    "prettier",              # YAML, JSON, etc linting
+    "doc8"                   # RST linting
+]
+test = [
+    "pytest",		 # Required for unit tests
+    "pytest-cov",    # Reports on test coverage
+    "pixelmatch"
 ]
 debug = [
-    "ipython",
+    "ipython"
 ]
 
-[project.entry-points."geoips.plugin_packages"]
+[tool.poetry.plugins."geoips.plugin_packages"]
```

Comment:
Does tool.poetry.plugins work the same as project.entry-points?

Reply:
Generally, yes, this does work the same as entry points, but notice that the console-script entry points became `[tool.poetry.scripts]`. 

So, there are some difference between Poetry's handling of entry points and Setuptools, but the differences are trivial.

Reply:
yes